src/main/java/edu/harvard/iq/dataverse/Template.java

```diff
@@ -256,26 +268,31 @@ public void setMetadataValueBlocks() {
         metadataBlocksForView.clear();
         metadataBlocksForEdit.clear();
         List<DatasetField> filledInFields = this.getDatasetFields(); 
+
+        Map<String, String> instructionsMap = getInstructionsMap();
         
         List <MetadataBlock> viewMDB = new ArrayList<>();
-        List <MetadataBlock> editMDB=this.getDataverse().getMetadataBlocks(true);
+        List <MetadataBlock> editMDB=this.getDataverse().getMetadataBlocks(false);
```

Comment:
Note this line is a fix for a bug found in testing that appears to come from #8837. With true, this call only gets metadatablocks from the current collection and, if it isn't the metadatablockroot/inherits which blocks to use from its parent(s), then no blocks are returned. Using false (or getMetadataBlocks() which redirects to getMetadataBlocks(false)) the call recurses up through parents until it finds the one where the metadatablocks are defined.

Testing should check the cases from #8735 to make sure this doesn't revert anything. As far as I can tell, selecting a new block in a parent, or in the current collection and then editing an existing template does show the newly selected block as well.